packages/@react-aria/virtualizer/src/ScrollView.tsx

```diff
@@ -148,7 +150,25 @@ function ScrollView(props: ScrollViewProps, ref: RefObject<HTMLDivElement>) {
   useLayoutEffect(() => {
     updateSize();
   }, [updateSize]);
-  useResizeObserver({ref, onResize: updateSize});
+  let raf = useRef<ReturnType<typeof requestAnimationFrame> | null>();
+  let onResize = () => {
+    if (isOldReact) {
+      raf.current ??= requestAnimationFrame(() => {
```

Comment:
React 16 and 17 don't batch like 18. We instead emulate that (to a degree) by waiting for a raf before submitting our state changes